impl/src/bin/cargo-raze.rs

```diff
@@ -181,52 +192,34 @@ fn write_to_file(path: &Path, contents: &str, verbose: bool) -> Result<()> {
   Ok(())
 }
 
-fn calculate_workspace_root(
-  path: &str,
-  output_override: Option<&str>,
-  new_behavior: bool,
-) -> Result<PathBuf> {
-  // Default to the current directory '.'
-  let mut prefix_path: PathBuf = std::env::current_dir()?;
-
-  // Allow the command line option to take precedence
-  match output_override {
-    Some(output) => {
-      prefix_path.clear();
-      prefix_path.push(output);
-    },
-    None => {
-      if new_behavior {
-        if let Some(workspace_root) = find_bazel_workspace_root() {
-          prefix_path.clear();
-          prefix_path.push(workspace_root);
-          prefix_path.push(
-            path
-              // Remove the leading "//" from the path
-              .trim_start_matches('/'),
-          );
-        }
-      }
-    },
-  }
-
-  Ok(prefix_path)
+/* A helper struct representing the inputs specific to the Remote genmode */
+struct RemoteGenModeInputs<'settings> {
+  pub override_lockfile: Option<PathBuf>,
+  pub binary_deps: Option<&'settings HashMap<String, cargo_toml::Dependency>>,
 }
 
-#[test]
-fn test_calculate_workspace_root() {
-  assert_eq!(
-    calculate_workspace_root(&"path_a".to_string(), None, true).unwrap(),
-    std::env::current_dir().unwrap()
-  );
-
-  assert_eq!(
-    calculate_workspace_root(
-      &"path_b".to_string(),
-      Some(&"direct/path".to_string()),
-      true
-    )
-    .unwrap(),
-    std::path::Path::new("direct").join("path")
-  );
+/** Helper function for gathering inputs for `genmode = "Remote"` builds */
+fn gather_remote_genmode_inputs<'settings>(
+  bazel_root: &Path,
+  settings: &'settings RazeSettings,
+) -> RemoteGenModeInputs<'settings> {
+  if settings.genmode == GenMode::Remote {
```

Comment:
Nit: This is cleaner structured as an early return:

```rust
if settings.genmode != GenMode::Remote {
    return RemoteGenModeInputs {
      override_lockfile: None,
      binary_deps: None,
    }
}
// the rest

Reply:
Updated